### PR TITLE
Fix pre-commit and revise style

### PR DESCRIPTION
**Pull request by davide-f**
## Changes proposed in this Pull Request
Fix to pre-commit.
The pre-commit has also been partially aligned to the one by pypsa-eur

I feel that this may conflict with other PRs, including https://github.com/pypsa-meets-earth/pypsa-earth/pull/701
#701 shall be first reviewed and merged before fixing this.

Everything can be done in few hours

### CODE REVIEW

**Review by pz-max — APPROVED** [review]

File: config.tutorial.yaml [hunk 1]
```diff
@@ -337,8 +337,8 @@ monte_carlo:
       # Key: add below the pypsa object for the monte_carlo sampling, "network" is only allowed for filtering!
       # Value: currently supported format [l_bound, u_bound] or empty [], represent multiplication factors for the object
     loads_t.p_set: [0.9, 1.1]
-    generators_t.p_max_pu.loc[:, n.generators.carrier == "wind"]: [0.9, 1.1]
-    generators_t.p_max_pu.loc[:, n.generators.carrier == "solar"]: [0.9, 1.1]
+    # generators_t.p_max_pu.loc[:, n.generators.carrier == "wind"]: [0.9, 1.1]
```
Comment (pz-max) [inline config.tutorial.yaml:340]: @davide-f we need at least two rows to test the Monte-Carlo stuff. Test is currently failing because of that

**Review by pz-max — CHANGES_REQUESTED** [review]

File: config.tutorial.yaml (hunk 1 above)
Comment (pz-max) [reply config.tutorial.yaml:340]: added a commit to fix this... let's see

**Comment by pz-max** [comment]
pre-commit is failing with `exit-code: 3`:
The [documentation](https://github.com/PyCQA/docformatter/blob/b54a22caae3300f9c37f8c48666a5808d7ff8440/docs/source/usage.rst#use-from-the-command-line) says:
```
Possible exit codes from docformatter:

    1 - if any error encountered
    2 - if it was interrupted
    3 - if any file needs to be formatted (in --check or --in-place mode)
```

I think because of the recent PR that we merged, @davide-f you need to run locally the pre-commit and add the fixes that the **docformatter** is suggesting. Can you do that?

**Comment by davide-f** [comment]
> if any file needs to be formatted (in --check or --in-place mode)

@pz-max the yaml block of the pre-commit is failing because it cannot handle "generators_t.p_max_pu.loc[:, n.generators.carrier == "wind"]: [0.9, 1.1]"

Therefore, I commented these two lines in the tutorial and default configs.

That lead instead to issues in the CI.
I think two approaches may apply:
1. fix the monte carlo so to be able to work without those 2 lines, this is I think the best way, but it means that those lines are not tested in the workflow
2. revise the monte-carlo config test, in the test folder, to add those two lines and ignore that file when doing the pre-commit CI, as otherwise it leads to issues

Since @pz-max, you worked in the monte carlo, what is your feeling there?

File: config.tutorial.yaml (hunk 1 above)
Comment (davide-f) [reply config.tutorial.yaml:340]: I recreated the PR and kept this line commented out.
pretty YAML crashes with checking those lines. In this PR, there is a fix to monte_carlo caospy to make sure that it works even with a single parameter in the list of monte carlos. The other options have been tested as well.

**Comment by davide-f** [comment]
@pz-max The PR has been rewritten and force-pushed with the following changes:
1. The lines related to `generators_t.p_max_pu.loc ....` in monte_carlo have been commented out as pretty-yaml crashed in analysing them
2. The monte_carlo function has been revised to make sure that the function works with all the montecarlo generating options even when a single uncertain parameter is handled, in this case being the load demand

**Comment by davide-f** [comment]
@pz-max PEP 257 is there, if CI passes this PR can be merged. Locally seems ok

**Pull request merged**